go-peer/main.go

```diff
@@ -168,27 +167,29 @@ func main() {
 
 		opts = append(opts,
 			libp2p.Transport(ws.New, ws.WithTLSConfig(&tls.Config{Certificates: certs})),
-			libp2p.ListenAddrStrings("/ip4/0.0.0.0/tcp/0/ws", "/ip6/::/tcp/0/ws"),
+			libp2p.ListenAddrStrings("/ip4/0.0.0.0/tcp/0/ws"),
 		)
 	}
 
-	if len(announceAddrs) > 0 {
-		var announce []multiaddr.Multiaddr
-		for _, addr := range announceAddrs {
-			announce = append(announce, multiaddr.StringCast(addr))
+	addressFactory := func(addrs []multiaddr.Multiaddr) []multiaddr.Multiaddr {
+		addr := *announceAddr
+		if len(addr) != 0 {
+			// printErr("Announce address: %s\n", addr)
+			maddr := multiaddr.StringCast(addr)
+			addrs = append(addrs, maddr)
+			return addrs
 		}
-		opts = append(opts, libp2p.AddrsFactory(func([]multiaddr.Multiaddr) []multiaddr.Multiaddr {
-			return announce
-		}))
+		return addrs
 	}
 
 	opts = append(opts,
 		libp2p.Identity(privk),
 		libp2p.Transport(quicTransport.NewTransport),
 		libp2p.Transport(tcpTransport.NewTCPTransport),
 		libp2p.Transport(webtransport.New),
-		libp2p.ListenAddrStrings("/ip4/0.0.0.0/udp/9091/quic-v1", "/ip4/0.0.0.0/udp/9092/quic-v1/webtransport", "/ip4/0.0.0.0/tcp/9090"),
-		),
+		libp2p.ListenAddrStrings("/ip4/0.0.0.0/udp/9095/quic-v1", "/ip4/0.0.0.0/udp/9096/quic-v1/webtransport"),
```

Comment:
It's best practice to run QUIC and WebTransport on the same port. With the v0.27.1 release, it also allows us to infer our WebTransport public address from QUIC observations.

```suggestion
		libp2p.ListenAddrStrings("/ip4/0.0.0.0/udp/9095/quic-v1", "/ip4/0.0.0.0/udp/9095/quic-v1/webtransport"),
```